mne_bids/write.py

```diff
@@ -1249,39 +1263,111 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
     _participants_json(participants_json_fname, True, verbose)
     _scans_tsv(raw, op.join(kind, bids_fname), scans_fname, overwrite, verbose)
 
-    # TODO: Implement coordystem.json and electrodes.tsv for EEG
-    electrodes_fname = make_bids_basename(
-        subject=subject_id, session=session_id, acquisition=acquisition,
-        suffix='electrodes.tsv', prefix=data_path)
+    # for MEG, we only write coordinate system
     if kind == 'meg' and not emptyroom:
         _coordsystem_json(raw, unit, orient,
                           manufacturer, coordsystem_fname, kind,
                           overwrite, verbose)
-    elif kind == 'ieeg':
-        coord_frame = "mri"  # defaults to MRI coordinates
-        unit = "m"  # defaults to meters
-    else:
-        coord_frame = None
-        unit = None
 
+    # write electrodes data for iEEG and EEG
+    electrodes_fname = make_bids_basename(
+        subject=subject_id, session=session_id, acquisition=acquisition,
+        suffix='electrodes.tsv', prefix=data_path)
+    unit = "m"  # defaults to meters
     # We only write electrodes.tsv and accompanying coordsystem.json
     # if we have an available DigMontage
     if raw.info['dig'] is not None:
         if kind == "ieeg":
+            # get coordinate frame from digMontage
+            digpoint = raw.info['dig'][0]
+            if any(digpoint['coord_frame'] != _digpoint['coord_frame']
+                   for _digpoint in raw.info['dig']):
+                warn("Defaulting coordinate frame of iEEG coords to "
+                     "'mri' since not all digpoints "
+                     "have the same coordinate frame.")
+                coord_frame = "mri"  # defaults to MRI coordinates
+            else:
+                # get the accepted mne-python coordinate frames
+                coord_frame_int = int(digpoint['coord_frame'])
+                coord_frame = _verbose_ieeg_coordinate_frames.\
+                    get(coord_frame_int, None)
+
+                # default coordinate frame to mri if not available
+                if coord_frame is None:
+                    coord_frame = 'mri'
+
             coords = _extract_landmarks(raw.info['dig'])
 
-            # Rescale to MNE-Python "head" coord system, which is the
-            # "ElektaNeuromag" system (equivalent to "CapTrak" system)
+            # RPA, NAS and LPA shouldn't be set in iEEG
             if set(['RPA', 'NAS', 'LPA']) != set(list(coords.keys())):
                 # Now write the data to the elec coords and the coordsystem
                 _electrodes_tsv(raw, electrodes_fname,
                                 kind, overwrite, verbose)
                 _coordsystem_json(raw, unit, orient,
                                   coord_frame, coordsystem_fname, kind,
                                   overwrite, verbose)
```

Comment:
I don't understand this. So we _only_ write if not all fiducials are present in the data? But we _do_ write if, say, only `LPA` and `RPA` are present (and `NAS` is missing)? Why can we not write regardless of whether fiducials are present in the data or not? If they don't make sense or are not required for iEEG data, they can be just omitted. In either case we should raise if we're not writing data, no?

Reply:
Sure that's fair, I left this in from awhile back in a different PR. I agree it really shouldn't matter if LPA, or RPA, NAS are there or not for iEEG.